src/Operations.jl

```diff
@@ -2107,17 +2107,17 @@ function print_status(env::EnvCache, old_env::Union{Nothing,EnvCache}, registrie
         tipend = manifest ? " -m" : ""
         tip = show_usagetips ? " To see why use `status --outdated$tipend`" : ""
         if !no_packages_upgradable && no_visible_packages_heldback
-            printpkgstyle(io, :Info, "Packages marked with $upgradable_indicator have new versions available", color=Base.info_color(), ignore_indent)
+            printpkgstyle(io, :Info, "Packages marked with $upgradable_indicator have new versions available and may be upgradable.", color=Base.info_color(), ignore_indent)
         end
         if !no_visible_packages_heldback && no_packages_upgradable
-            printpkgstyle(io, :Info, "Packages marked with $heldback_indicator have new versions available but cannot be upgraded.$tip", color=Base.info_color(), ignore_indent)
+            printpkgstyle(io, :Info, "Packages marked with $heldback_indicator have new versions available but compatibility constraints restrict them from upgrading.$tip", color=Base.info_color(), ignore_indent)
         end
         if !no_visible_packages_heldback && !no_packages_upgradable
-            printpkgstyle(io, :Info, "Packages marked with $upgradable_indicator and $heldback_indicator have new versions available, but those with $heldback_indicator cannot be upgraded.$tip", color=Base.info_color(), ignore_indent)
+            printpkgstyle(io, :Info, "Packages marked with $upgradable_indicator and $heldback_indicator have new versions available, but those with $heldback_indicator are restricted by compatibility constraints from upgrading.$tip", color=Base.info_color(), ignore_indent)
```

Comment:
```suggestion
            printpkgstyle(io, :Info, "Packages marked with $upgradable_indicator and $heldback_indicator have new versions available, but those with $heldback_indicator are known to be restricted by compatibility constraints from upgrading.$tip", color=Base.info_color(), ignore_indent)
```

Reply:
Seems like 3 extra words with not so much extra info. If it is restricted, it is also "known to be restricted"? :P

Reply:
I was hoping to help the user infer that the ^ might not be upgradable because of an unknown compat limit, which they might come across. How the language currently is seems to be definitive that only ⌅ is limited.

Reply:
I'll go with this for now and we can tweak it further later if people still get confused.